chat_server/src/main.py

```diff
@@ -1,153 +1,207 @@
             context.abort(
-                grpc.StatusCode.RESOURCE_EXHAUSTED,
-                f"Message queue for user {to_user} is full",
+                grpc.StatusCode.NOT_FOUND, f"User {to_user} is do not exist"
             )
-            logging.debug(context.details())
+            return chat_pb2.SendMessageReply()
+        handler_to_send.add_message_to_queue(
+            to_send_queue=True, value=MessageToJson(request.message)
+        )
+        handler_to_store.add_message_to_queue(
+            to_send_queue=False, value=MessageToJson(request.message)
+        )
+
         logging.debug(f"Message added to queue for user: {to_user}")
-        # TODO: Save message to date store
         return chat_pb2.SendMessageReply()
 
-    def RecieveMessages(self, request, context) -> chat_pb2.RecieveMessagesReply:
-        """Receive messages to user.
+    def RecieveMessages(
+        self, request: chat_pb2.RecieveMessagesRequest, context
+    ) -> chat_pb2.RecieveMessagesReply:
+        """Receives messages to user.
 
         When connection is active, takes messege from users queue and yield it
         till connected client get it.
 
         Also every 30 seconds yield empty message to help sinchronize client with server.
 
-        Parameters
-        ----------
-        request : chat_bp2.RecieveMessagesRequest
-            Request defined in chat.proto file
-        context : grpc context
-
-        Returns
-        -------
-        chat_pb2.RecieveMessagesReply
-            Reply defined in chat.proto file
-
-        Grpc_errors
-        -------
-        grpc.StatusCode.UNAUTHENTICATED
-            When user who want to listen doesn't exist
+        Args:
+            request: Request defined in chat.proto file.
+            context: grpc context.
+
+        Returns:
+            chat_pb2.RecieveMessagesReply: Reply defined in chat.proto file.
+
+        Yields:
+            Iterator[chat_pb2.RecieveMessagesReply]: Iterate on reply defined in chat.proto file.
+
+        Raises grpc_error
+            grpc.StatusCode.UNAUTHENTICATED: When user who want to listen doesn't exist.
         """
         stream_to_user = request.to_user_login
-        if not self._check_user(stream_to_user):
+        try:
+            handler = EtcdMessagesHandler(
+                client=self.etcd_client, to_user=stream_to_user
+            )
+        except KeyError as e:
             context.abort(
                 grpc.StatusCode.UNAUTHENTICATED,
                 f"User {stream_to_user} is not registred",
             )
             return chat_pb2.RecieveMessagesReply()
 
-        start = time.time()
+        response = handler.get_elems_from_queue(
+            from_send_queue=False, get_all=True
+        )
+        for _, elem in response[-10:]:
+            message = Parse(elem, chat_pb2.Message())
+            yield chat_pb2.RecieveMessagesReply(message=message)
+        logging.debug("[10 messeges from previous session restored]")
```

Comment:
Why do we need in square brackets here?

Reply:
Right, there is no need for that, done in latest commit